libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs

```diff
@@ -282,12 +277,8 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
         /// the result contains the user's token.</remarks>
         public async Task<TokenResponse> GetUserTokenAsync(DialogContext dc, CancellationToken cancellationToken = default(CancellationToken))
         {
-            if (!(dc.Context.Adapter is IUserTokenProvider adapter))
-            {
-                throw new InvalidOperationException("OAuthPrompt.GetUserToken(): not supported by the current adapter");
-            }
-
-            return await adapter.GetUserTokenAsync(dc.Context, ConnectionName.GetValue(dc.State), null, cancellationToken).ConfigureAwait(false);
+            var settings = new OAuthPromptSettings { ConnectionName = ConnectionName?.GetValue(dc.State) };
```

Comment:
If connection name does not exist, then none of this really makes sense. What is the scenario where ConnectionName is null? Should we fail fast in that scenario?

Reply:
All code paths throw an ArgumentNullException exception on ConnectionName before attempting to call the service. I think that seems reasonable for the code-first scenario (it being clear from the exception what is going on.)

But for the property based adaptive scenario is there an advantages to validating the properties are set (shouldn't that be done by schema?)

Reply:
we think we're OK with the argument checkin on the next layer